### DISCUSSION TITLE
Dedicated Fleet Node not creating a new Fleet server in fleet

### DISCUSSION THREAD

**Discussion by jaypfe** [Security-Onion-Solutions/securityonion#11530] [2.4] [answered]
### Version

2.4.20

### Installation Method

Security Onion ISO image

### Description

configuration

### Installation Type

Distributed

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### CPU

8

### RAM

16

### Storage for /

128

### Storage for /nsm

128

### Network Traffic Collection

other (please provide detail below)

### Network Traffic Speeds

Less than 1Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

I spun up a fleet node to put in my DMZ. It joined the grid okay, and I can see the machine got added as an agent under the grid_nodes_general policy. But a separate fleet server was never added. I can see a policy was created, just no fleet server. 

I have tried twice, powering off machine, removing from the grid nodes, but same result. 
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/107776997/7c136941-be69-4351-84f5-415be8e16a57)

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by defensivedepth** [comment]
Hello there!

`I have tried twice, powering off machine, removing from the grid nodes, but same result.`

What do you mean when you say `removing from the grid nodes`?

Do you see the Fleet Server in the Grid page in the Security Onion Console? Does it show any errors?

**Reply by jaypfe** [reply]
Yes I see the fleet server appear in the grid page, with no faults. 
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/107776997/5c402095-5dab-411c-aa1c-f3776f270d6d)
The first time I tried, I had this issue. So I removed the node from the grid members page. I then spun up a new fleet node and joined it. 

Everything looks good and worked exactly how it should have except for the new fleet server not being created on the fleet page. Its odd.

**Reply by jaypfe** [reply]
As a side note, the device never left influxDB after removing from the Grid
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/107776997/300efb41-76d1-4d3c-8a14-e51f1b5b2ddd)

**Reply by defensivedepth** [reply]
If you drill down into the Node in the Grid page, do you see the Fleet server container running?

Can you take a screenshot of the `Settings` page in the Fleet Management UI? (shows the Fleet URLs etc)

**Reply by jaypfe** [reply]
Absolutely. Here is that grid nodes:
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/107776997/bda3ac8d-4e48-457f-9a24-55107662abcf)

Settings page:
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/107776997/320a552b-1ee2-4343-8248-143d9a3e4256)

It is my understanding that another fleet server would be visible on the fleet page along with the actual node its self in the grid_nodes_general just like how it happens on the manager, Is that correct? Or do I have it completely wrong?

**Reply by defensivedepth** [reply]
Ya there should be another Fleet server showing up in the Fleet page.

If you drop down the `Status` filter and select `inactive` and `unenrolled`, does it show up there?

Also, if you run the following from the command line on the Fleet node, doe you see any errors?

`sudo so-elastic-fleet-restart --force`

**Reply by jaypfe** [reply]
No errors with that command.  But afterwards the container went missing for a bit, but it did come back
I have also observed a steady increase in ram usage up to about 92% slowly day over day of 16gb before I restarted. 

So interestingly when I tried the first time to add a fleet node, it added another fleet server tied to the manager. The status of the original manager fleet server went inactive so I un-enrolled it. Then I tried again to add a new fleet node but I did not see that behavior replicated. Just the addition of the new node under the grid nodes policy. 
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/107776997/9dbd7cb5-9715-42d5-9440-a499ea00b637)

Does this work for you guys in house? I am not against trying again with a new node.  I am not absolutely against redeploying again either, but would like to be reasonably certain that would fix it lol. 

It is hosted in my DMZ, I have these ports open to it
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/107776997/a285cbd9-71fe-429f-a703-73584ba5424b)

Security onion 04 is 8443

**Reply by defensivedepth** [reply]
So the Fleet Servers run in containers and store their state in Elasticsearch. So if a container is removed & replaced (normal thing that happens when so-elastic-fleet-restart is run), the new container enrolls as a new fleet server - this is by design, directly from Elastic. So you will see older fleet servers in the Fleet listing, but they will go inactive and be hidden from the default view. 

It's possible that manually unenrolling it caused the enrollment key to expire. 

Try running the following on the Fleet Node:

`sudo docker logs so-elastic-fleet > logs`
`cat logs`

Any errors from those logs? It should give you the enrollment status.

And yes, we have a long-running internal distributed grid with a dedicated fleet node working as expected. As well as nightly automated testing for various architectures, including this one.

**Reply by jaypfe** [reply]
There are several of these errors: 
{"log.level":"error","@timestamp":"2023-10-20T14:48:56.897Z","message":"fail elasticsearch info","component":{"binary":"fleet-server","dataset":"elastic_agent.fleet_server","id":"fleet-server-default","type":"fleet-server"},"log":{"source":"fleet-server-default"},"cluster.addr":["tul-so-manager01:9200"],"cluster.maxConnsPersHost":128,"error.message":"dial tcp xx.x.xx.xxx:9200: i/o timeout","ecs.version":"1.6.0","service.name":"fleet-server","ecs.version":"1.6.0"}

Does 9200 need to be open between this node and the manager? I did not see that port required in the docs.

**Reply by defensivedepth** [reply, marked as answer]
Ah yes, I missed that in your port list screencap. Yes, the Fleet server needs to be able to access ES (TCP/9200) on the Manager. 

I will get the docs updated.

**Reply by jaypfe** [reply]
Sweeeeeet. I will make that change. I will let update if it works just with that, or if it needs a new node spun in place.
Thanks for your help on this one Josh,

**Reply by defensivedepth** [reply]
Let me know how it goes and if there are other issues that you run into. I think the Fleet node + Elastic Agents are a powerful addition to SO and we want to make it as usable as possible.

**Reply by jaypfe** [reply]
After restarting, I still see that it cannot connect. I will try to spin up a new node and see what it does.

**Reply by jaypfe** [reply]
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/107776997/6e429d48-adfc-41eb-9585-987a0cd21a0b)
There she be!

**Reply by defensivedepth** [reply]
Fantastic!

**Reply by defensivedepth** [reply]
btw, what specific docs were you looking at that doesnt include TCP/9200

**Reply by jaypfe** [reply]
I was looking here in the firewall section: 
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/107776997/f6fddb7f-e025-48df-b6a5-27633ae8c692)

https://docs.securityonion.net/en/2.4/firewall.html?highlight=ports

**Reply by jaypfe** [reply]
@defensivedepth 
I did not want to open a new thread as I think its related, but I am having a new issue.  But I can if that is what is preferred! 

In the fleet interface, the new DMZ fleet node has gone inactive, with no new active one taking place. 

I also have a server in the DMZ that I am collecting logs from, that shows inactive as well. This one has to go through the DMZ to get back to the manager. 

The kicker is its still sending logs, at the same rate as I would expect, but showing offline. That means the fleet node has to still be receiving and the server is in fact online.

In the fleet logs section that one DMZ sever is erroring with these messeges:  
[elastic_agent][error] Cannot checkin in with fleet-server, retrying  
This is over and over and over. 
I can see the fleet server creating logs in hunt as well, though not as much as the manager fleet server (which is doing much more). 

Looking at the DMZ fleet server in the fleet logs area I can see it saying: [elastic_agent][error] ack retrier: commit failed with error: acknowledge 1 actions '[action_id: policy:FleetServer_dmz-so-fleet01:24:1, type: POLICY_CHANGE]' for elastic-agent '18ffbef4-f337-4007-84d9-39e8541c7fdd' failed: fail to ack to fleet: all hosts failed: 1 error occurred:
	* requester 0/1 to host https://localhost:8221/ errored: Post "https://localhost:8221/api/fleet/agents/18ffbef4-f337-4007-84d9-39e8541c7fdd/acks?": dial tcp 127.0.0.1:8221: connect: connection refused

It was saying that pretty often yesterday, but I have not seen it print any more logs of any kind to the fleet log area. 

But yet, its still sending logs? Any idea? 

Maybe another port I need to open haha!?

**Reply by jaypfe** [reply]
running the log command you gave me before showed it was still in enrolled 
I ran sudo so-elastic-fleet-restart --force with no errors 

After doing that a new fleet server appeared in fleet, and I was able to enroll that new agent. The existing agent that was still sending logs, but showing offline has now come back online.

**Reply by defensivedepth** [reply]
Just confirming that your standalone Fleet Node has DNS setup correctly so that it can resolve the Manager hostname etc?

**Reply by jaypfe** [reply]
Yes, all that is good. 

It came back after running 'sudo so-elastic-fleet-restart --force' as listed above. 
Pretty odd behavior that it was still trafficking logs from the agent that had already registered, but showing that agent offline and itself as inactive and could not register any new agents

But after running that command, a new fleet server came up as healthy, and everything worked as expected after that. 

It might be worth noting as well, I restarted the entire grid before running that command and it did not seem to have any effect.

**Reply by defensivedepth** [reply]
FYI, restart of the grid wont replace the containers. Just stops & starts them. `so-elastic-fleet-restart` stops & rm the container and creates a new container from the image.

**Reply by jaypfe** [reply]
ahhhh got it. Thanks again man!

I should not expect to need to run that command again right? 

do you happen to have some troubleshooting steps with that in the event it happens again?
